clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/KafkaApplicationClientFactory.java

```diff
@@ -14,10 +14,82 @@
 package org.eclipse.hono.application.client.kafka;
 
 import org.eclipse.hono.application.client.ApplicationClientFactory;
+import org.eclipse.hono.application.client.DownstreamMessage;
+import org.eclipse.hono.application.client.MessageConsumer;
+import org.eclipse.hono.client.kafka.consumer.KafkaConsumerCommitException;
+import org.eclipse.hono.client.kafka.consumer.KafkaConsumerPollException;
+
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
 
 /**
  * A factory for creating clients for Hono's Kafka-based northbound APIs.
  */
 public interface KafkaApplicationClientFactory extends ApplicationClientFactory<KafkaMessageContext> {
 
+    /**
+     * Creates a client for consuming data from Hono's north bound <em>Telemetry API</em>.
+     * <p>
+     * The messages passed in to the consumer will be acknowledged automatically if the message handler does not throw
+     * an exception. <b> NB: Be very careful about throwing exceptions in the message handler, after that no more
+     * telemetry messages will be consumed for the client until the problem is fixed!</b>
+     * <p>
+     * If a fatal error occurs, the consumer will be closed and the close-handler, if it is not {@code null}, invoked
+     * with an exception indicating the cause. There are error cases that might disappear later on and where it makes
+     * sense to create a new consumer and other cases that need to be resolved externally.
+     * <p>
+     * </p>
+     * ERROR CASES:
+     * <p>
+     * Errors can happen when polling, in message processing, and when committing the offset to Kafka. If a {@code poll}
+     * operation fails, the consumer will be closed and the close handler will be passed a
+     * {@link KafkaConsumerPollException} indicating the cause. If the message processing fails, the consumer will be
+     * closed and the exception will be passed to the close handler. If the offset commit fails, the consumer will be
```

Comment:
it is unclear to me here, what you are referring to with *message processing fails*. Is this about the provided handler throwing an exception or is this referring to processing within the kafka-client? In any case, is there a specific type of exception being passed in to the close handler?

Reply:
> Is this about the provided handler throwing an exception or is this referring to processing within the kafka-client?

The former one. It is the exception thrown by the message handler. Do you see value in wrapping this exception?

Reply:
No, and I also do not understand why the close handler will be invoked in this case in the first place. But if this is really necessary, then the wording should be changed to better describe which exception it is that is passed in. At least I did not understand without your explaining ...

Reply:
I changed it so that the close handler is no longer invoked when the message handler throws an exception.